### PR TITLE
Littlenavmap

### PR DESCRIPTION
**Pull request by merspieler**
## Description of changes

Add little nav map and it's dependency atools
Little nav map is a free open source flight planner, navigation tool, moving map, airport search and airport information system

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/at/atools/package.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{ fetchFromGitHub
+, pkg-config
+, stdenv
+, lib
+# Package dependencies
+, libsForQt5
```
Comment (auroraanna) [inline pkgs/by-name/at/atools/package.nix:6]:
```suggestion
, qt5
```
this works too, and it's shorter and it's the way going forward with qt6

**Review by auroraanna — CHANGES_REQUESTED** [review]

**Comment by auroraanna** [comment]
uhh… there's a bunch of non-executable stuff in `$out/bin` which shouldn't be there. is the program exprecting these files in the same directory and subdirectories?

File: pkgs/by-name/at/atools/package.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    libsForQt5.qtwebengine
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/albar965/atools";
+    description = "static library extending Qt for exception handling,
+a log4j like logging framework, Flight Simulator related utilities like BGL reader
+and more.";
+    license = licenses.asl20;
+    platforms = platforms.linux;
```
[the remark this answers is not in the document]
Comment (auroraanna) [reply pkgs/by-name/at/atools/package.nix:44]: ofborg can compile on 4 platforms afaik

[the remark this answers is not in the document]
Comment (auroraanna) [reply pkgs/by-name/at/atools/package.nix:44]: well setting it to linux prevents people on the other platforms from trying it out though and reporting if it works or not. did you test aarch64-linux e.g.?

[the remark this answers is not in the document]
Comment (auroraanna) [reply pkgs/by-name/at/atools/package.nix:44]: at least if they don't want to clone nixpkgs and change the platform

File: pkgs/by-name/li/littlenavmap/package.nix [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+let
+customMarble = plasma5Packages.marble.overrideAttrs (old: { src = fetchFromGitHub {
+      owner = "albar965";
+      repo = "marble";
+      rev = "722acf7f8d79023f6c6a761063645a1470bb3935"; # MUST be lnm/1.1 branch
+      sha256 = "sha256-5GSa+xIQS9EgJXxMFUOA5jTtHJ6Dl4C9yAkFPIOrgo8=";
+    }; preConfigure = ''
+    cmakeFlags+=" -DQTONLY=TRUE -DBUILD_MARBLE_EXAMPLES=NO -DBUILD_INHIBIT_SCREENSAVER_PLUGIN=NO -DBUILD_MARBLE_APPS=NO -DBUILD_MARBLE_EXAMPLES=NO -DBUILD_MARBLE_TESTS=NO -DBUILD_MARBLE_TOOLS=NO -DBUILD_TESTING=NO -DBUILD_WITH_DBUS=NO -DMARBLE_EMPTY_MAPTHEME=YES -DMOBILE=NO -DWITH_DESIGNER_PLUGIN=NO -DWITH_Phonon=NO -DWITH_Qt5Location=NO -DWITH_Qt5Positioning=NO -DWITH_Qt5SerialPort=NO -DWITH_ZLIB=NO -DWITH_libgps=NO -DWITH_libshp=NO -DWITH_libwlocate=NO -Wno-deprecated-copy -Wno-deprecated -Wno-deprecated-declarations -DINCLUDE_INSTALL_DIR=''${!outputDev}/include"
+  ''; });
```
[the remark this answers is not in the document]
Comment (auroraanna) [reply pkgs/by-name/li/littlenavmap/package.nix]: still confusing indentation. `preConfigure` inside `customMarble` but it has the same indentation as `customMarble`.

**Comment by auroraanna** [comment]
*suggests https://github.com/auroraanna/nixpkgs/tree/littlenavmap-anna to you* i made this branch because i can't add patch files in the github suggestions

**Comment by auroraanna** [comment]
just realized it probably only runs on linux because of this package expecting Wayland/X11 and freedesktop

File: pkgs/by-name/at/atools/package.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  buildInputs = [
+    qt5.qtwebengine
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/albar965/atools";
+    description = "static library extending Qt for exception handling,
+a log4j like logging framework, Flight Simulator related utilities like BGL reader
+and more.";
+    license = licenses.asl20;
```
Comment (auroraanna) [inline pkgs/by-name/at/atools/package.nix:43]: atools is GPL too…
```suggestion
    license = licenses.gpl3Plus;
```

**Comment by auroraanna** [comment]
> just realized it probably only runs on linux because of this package expecting Wayland/X11 and freedesktop

[atools though has build instructions for several platforms](https://github.com/albar965/atools/blob/master/BUILD.txt) and is a library

**Review by auroraanna — APPROVED** [review]

File: pkgs/by-name/at/atools/package.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+stdenv.mkDerivation rec {
+  pname = "atools";
+  version = "4.0.6";
+
+  src = fetchFromGitHub {
+    owner = "albar965";
+    repo = "atools";
+    rev = "v${version}";
+    sha256 = "sha256-ieL7cGqXNmGMyTf8tXCSmwQFwyHyy8NwGJ+3Zg92nmM=";
```
Comment (kirillrdy) [inline pkgs/by-name/at/atools/package.nix:17]:
```suggestion
    hash = "sha256-ieL7cGqXNmGMyTf8tXCSmwQFwyHyy8NwGJ+3Zg92nmM=";
```

https://nixos.org/manual/nixpkgs/unstable/#fetchfromgithub

File: pkgs/by-name/at/atools/package.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{ fetchFromGitHub
+, pkg-config
+, stdenv
+, lib
+# Package dependencies
```
Comment (kirillrdy) [inline pkgs/by-name/at/atools/package.nix:5]: any reason for this comment ?

File: pkgs/by-name/li/littlenavmap/package.nix [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+, atools
+, plasma5Packages
+}:
+
+let
+  customMarble = plasma5Packages.marble.overrideAttrs (old: { src = fetchFromGitHub {
+      owner = "albar965";
+      repo = "marble";
+      rev = "722acf7f8d79023f6c6a761063645a1470bb3935"; # MUST be lnm/1.1 branch
+      sha256 = "sha256-5GSa+xIQS9EgJXxMFUOA5jTtHJ6Dl4C9yAkFPIOrgo8=";
```
Comment (kirillrdy) [inline pkgs/by-name/li/littlenavmap/package.nix:16]:
```suggestion
      hash = "sha256-5GSa+xIQS9EgJXxMFUOA5jTtHJ6Dl4C9yAkFPIOrgo8=";
```

File: pkgs/by-name/li/littlenavmap/package.nix [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+    atools
+    customMarble
+  ];
+
+  buildPhase = ''
+    runHook preBuild
+    mkdir -p build
+    cd build
+    qmake ../littlenavmap.pro CONFIG+="release" QMAKE_CXXFLAGS+=' -Wno-deprecated-copy -Wno-deprecated -Wno-deprecated-declarations'
+    make -j $(nproc)
```
Comment (kirillrdy) [inline pkgs/by-name/li/littlenavmap/package.nix:58]:
```suggestion
    make -j $NIX_BUILD_CORES
```

File: pkgs/by-name/li/littlenavmap/package.nix [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+  src = fetchFromGitHub {
+    owner = "albar965";
+    repo = "littlenavmap";
+    rev = "v${version}";
+    sha256 = "sha256-u3H2U+8HZD18Yt6UB7QTTQOrrzN03PGolbFYddG8AIc=";
+  };
+
+  patches = [ ./littlenavmap.desktop.patch ];
+
+  ATOOLS_LIB_PATH = atools;
```
Comment (kirillrdy) [inline pkgs/by-name/li/littlenavmap/package.nix:36]: how does this work ?

Comment (kirillrdy) [reply pkgs/by-name/li/littlenavmap/package.nix:36]: ignore this, my bad

**Pull request closed without merging**